rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -850,13 +865,23 @@ def _predict_label(
         label_ids = message_sim.argsort()[::-1]
 
         if (
-            self.component_config[LOSS_TYPE] == SOFTMAX
+            self.component_config[LOSS_TYPE] == CROSS_ENTROPY
             and self.component_config[RANKING_LENGTH] > 0
         ):
-            message_sim = train_utils.normalize(
+            message_sim = train_utils.filter_top_k(
```

Comment:
> filtering was needed only for renormalization. Number of top k prediction is regulated later in lines 896-902
👍 

> I don't think normalization should be applied for `inner`

You mean `similarity_type=inner` or `model_confidence=`inner`. Currently, I am changing the if condition to this - 

```
if (
            self.component_config[LOSS_TYPE] == CROSS_ENTROPY
            and self.component_config[RANKING_LENGTH] > 0
            and self.component_config[SIMILARITY_TYPE] == INNER
            and self.component_config[MODEL_CONFIDENCE] == SOFTMAX
        ):
```
This would ensure that normalization is done only for older settings. In any of the new setting it won't be done.

Reply:
oh, did I misread `SIMILARITY_TYPE` and `MODEL_CONFIDENCE`? don't we have a check when loading params that `MODEL_CONFIDENCE] == SOFTMAX` only if other conditions meet? then I think here we can only check for `MODEL_CONFIDENCE] == SOFTMAX`

Reply:
Do you mean that `model_confidence` should be softmax only if - 
1. `loss_type` is `cross entropy`
2. `similarity_type` is `inner`.

We don't make these checks explicitly right now. I think it makes sense to add. Will do 👍